### PR TITLE
Clarify that `Approve` and `Reject` always exit

### PR DESCRIPTION
**Pull request by fabrice102**
I made this PR following https://forum.algorand.org/t/innertxnbuilder-second-transaction-is-not-stored-in-the-blockchain/8893/2?u=fabrice

### CODE REVIEW

**Review by ahangsu — APPROVED** [review]
lookin good! Thanks for the addition in documentation.

**Comment by ahangsu** [comment]
just a nit: adding a line for this PR in CHANGELOG https://github.com/algorand/pyteal/blob/master/CHANGELOG.md would be very helpful!

File: docs/control_structures.rst [hunk 1]
```diff
@@ -19,7 +19,8 @@ The :any:`Approve` and :any:`Reject` expressions cause the program to immediatel
 is used, then the execution is marked as successful, and if :code:`Reject` is used, then the execution
 is marked as unsuccessful.
 
-These expressions also work inside :ref:`subroutines <subroutine_expr>`.
+These expressions also work inside :ref:`subroutines <subroutine_expr>`. When used inside subroutines, they
+also cause the program to immediately exit, contrary to :code:`Return(...)` which just returns the subroutine.
```
Comment (jasonpaulos) [inline docs/control_structures.rst:23]:
```suggestion
also cause the program to immediately exit, contrary to :code:`Return(...)` which just returns from the subroutine.
```

**Comment by tzaffi** [comment]
@fabrice102 - nudge, in case you're back from vacay. Can you have a look at the suggested changes?

**Comment by fabrice102** [comment]
@tzaffi I've merged the change proposed by @jasonpaulos and added a line in the CHANGELOG.

**Pull request merged**